client/modules/IDE/components/PreviewFrame.jsx

```diff
@@ -84,22 +84,19 @@ class PreviewFrame extends React.Component {
 
   handleConsoleEvent(messageEvent) {
     if (Array.isArray(messageEvent.data)) {
-      messageEvent.data.forEach((message) => {
-        const args = message.arguments;
+      messageEvent.data.every((message, index, arr) => {
+        const { arguments: args } = message;
         Object.keys(args).forEach((key) => {
           if (typeof args[key] === 'string' && args[key].includes('Exiting potential infinite loop')) {
             this.props.stopSketch();
```

Comment:
if this is true, you'll probably want to return false, right? maybe you could have some variable outside of the forEach loop, and if it gets set because there is an infinite loop, you could exit from the `every` loop early.